### PR TITLE
chore(tests): migrate to vitest

### PR DESCRIPTION
**Pull request by arjunvegda**
## Summary
- Migrate to vitest

## Check List

- [x] `yarn run prettier` for formatting code and docs
- [x] Wait for #1823 to be merged + update tests if needed
- [x] Ensure that the old TypeScript tests work as expected

### CODE REVIEW

File: tests/babel/plugin-debug-label.test.ts [hunk 1]
```diff
@@ -13,28 +14,28 @@ const transform = (
     plugins: [[plugin, { customAtomNames }]],
   })?.code
 
-it('Should add a debugLabel to an atom', () => {
+it('should add a debugLabel to an atom', () => {
```
Comment (dai-shi) [inline tests/babel/plugin-debug-label.test.ts:17]: Let's have these changes not related to vitest migration in a separate PR.

Comment (arjunvegda) [reply tests/babel/plugin-debug-label.test.ts]: Sure, [done](https://github.com/pmndrs/jotai/pull/1839/commits/704c6313adb88f8f43c03b7bb337de64ba6185c1)!

File: package.json [hunk 2]
```diff
@@ -73,10 +73,11 @@
...
     "pretest": "tsc --noEmit",
-    "test": "yarn node --experimental-vm-modules $(yarn bin jest)",
-    "test:ci": "yarn node --experimental-vm-modules $(yarn bin jest)",
-    "test:dev": "yarn node --experimental-vm-modules $(yarn bin jest) --watch --no-coverage",
-    "test:coverage:watch": "yarn node --experimental-vm-modules $(yarn bin jest) --watch",
+    "test": "vitest",
+    "test:ui": "yarn run test --ui",
+    "test:coverage": "yarn run test run --coverage",
+    "test:ci": "yarn run test run",
+    "test:coverage:watch": "yarn run test --coverage",
```
Comment (dai-shi) [inline package.json:80]: Let's remove yarn dependency (which I added). I want to simplify scripts.
How about this?
```suggestion
    "test": "vitest --ui --coverage",
    "test:ci": "vitest",
```

File: tests/babel/plugin-debug-label.test.ts [hunk 3]
```diff
@@ -1,9 +1,6 @@
-import path from 'path'
 import { transformSync } from '@babel/core'
-import { expect, it } from '@jest/globals'
-
-// const plugin = path.resolve('./src/babel/plugin-debug-label.ts')
-const plugin = path.resolve('./dist/babel/plugin-debug-label.js')
+import { expect, it } from 'vitest'
+import plugin from 'jotai/babel/plugin-debug-label'
```
Comment (dai-shi) [inline tests/babel/plugin-debug-label.test.ts:3]: How nice!

File: .github/workflows/test-old-typescript.yml [hunk 4]
```diff
@@ -61,12 +61,19 @@ jobs:
           sed -i~ 's/"jsx": "react-jsx",/"jsx": "react",/' tsconfig.json
           sed -i~ 's/"noUncheckedIndexedAccess": true,//' tsconfig.json
           sed -i~ 's/^import type /import /' tests/*/*.tsx tests/*/*/*.tsx
-          yarn json -I -f package.json -e "this.resolutions={}; this.resolutions['@jest/globals']='25.5.2'; this.resolutions['pretty-format']='25.5.0'; this.resolutions['@types/prettier']='2.4.2'; this.resolutions['@types/node']='18.11.18';"
-          yarn add -D @jest/globals@25.5.2 pretty-format@25.5.0 @types/prettier@2.4.2 @types/node@18.11.18 @types/yargs@17.0.13 @types/babel__traverse@7.18.2
+          yarn json -I -f package.json -e "this.resolutions={};  this.resolutions['@types/prettier']='2.4.2'; this.resolutions['@types/node']='18.11.18';"
+          yarn add -D @types/prettier@2.4.2 @types/node@18.11.18 @types/yargs@17.0.13 @types/babel__traverse@7.18.2
           rm -r tests/react/vanilla-utils/atomWithObservable.*
+      - name: Install old TypeScript
```
Comment (dai-shi) [inline .github/workflows/test-old-typescript.yml:67]: I see. Install TS in advance and then patch.

**Review by dai-shi — APPROVED** [review]
Wow, the changes look very clean. Great work!

**Comment by dai-shi** [comment]
(I'm going to hold off merging until https://github.com/pmndrs/valtio/pull/646 is resolved.)

**Comment by dai-shi** [comment]
@arjunvegda Please merge main branch.

**Comment by arjunvegda** [comment]
> @arjunvegda Please merge main branch.

Done!

**Comment by rtritto** [comment]
Why _jest_ is replaced with _vitest_?

**Comment by dai-shi** [comment]
Jest's ESM support seems less stable than Vitest.

**Pull request merged**